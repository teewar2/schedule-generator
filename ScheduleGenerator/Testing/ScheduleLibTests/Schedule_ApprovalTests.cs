using System;
using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using static Infrastructure.SheetConstants;
using static Domain.DomainExtensions;

namespace Testing.ScheduleLibTests
{
    [TestFixture]
    [Explicit]
    [UseReporter(typeof(DiffReporter))]
    public class Schedule_ApprovalTests
    {
        [Test]
        public void CheckMeetingsPlaced_Approval()
        {
            var solver = GetSolver(SpringConfig, Repository);
            var schedule = solver
                .GetSolution(new(0, 0, 0, 0, 100))
                .Schedule;

            var placedMeetings = schedule
                .Meetings
                .Select(m => $"{m.Discipline} {m.MeetingType} {m.Teacher}")
                .OrderBy(m => m)
                .ToList();
            var notePlacedMeetings = schedule
                .NotUsedMeetings
                .Select(m => $"{m.Discipline} {m.MeetingType} {m.Teacher}")
                .OrderBy(m => m)
                .ToList();
            var unifiedString =
                $"Placed:{placedMeetings.Count} Left:{schedule.NotUsedMeetings.Count}, Placed:{Environment.NewLine}" +
                $"{string.Join(Environment.NewLine, placedMeetings)}{Environment.NewLine}" +
                $"Not placed:{Environment.NewLine}" +
                $"{string.Join(Environment.NewLine, notePlacedMeetings)}{Environment.NewLine}";
            Approvals.Verify(unifiedString);
        }
    }
}