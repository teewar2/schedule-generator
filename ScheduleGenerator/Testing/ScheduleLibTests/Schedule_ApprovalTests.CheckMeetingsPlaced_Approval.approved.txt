Placed:94 Left:0, Placed:
Алгем Lecture Волков Михаил
Алгем Lecture Волков Михаил
Алгем Seminar Белов
Алгем Seminar Белов
Алгем Seminar Белов
Алгем Seminar Белов
Алгем Seminar Бондарь
Алгем Seminar Бондарь
Алгем Seminar Гейн Павел
Алгем Seminar Гейн Павел
Алгем Seminar Косолобов
Алгем Seminar Косолобов
Алгем Seminar Липин Антон
Алгем Seminar Липин Антон
Алгем Seminar Михайлова
Алгем Seminar Михайлова
Алгем Seminar Расин
Алгем Seminar Расин
Алгоритмы ComputerLab Быков Алексей
Алгоритмы ComputerLab Когут Иван
Алгоритмы ComputerLab Субботин Игорь
Алгоритмы ComputerLab Трифочкин Семен
Алгоритмы Lecture Рубинчик, Косолобов, Меркурьев, Ипатов
АрхЭвм Lecture Пьянзин
БД ComputerLab Эйдинов, Шляхов, Пушкин
БД ComputerLab Эйдинов, Шляхов, Пушкин
Веб-разработка ComputerLab Митрофанова + Усков
Веб-разработка ComputerLab Митрофанова + Усков
Дизайн Seminar Пушин + преподаватель0
Дизайн Seminar Пушин + преподаватель0
Дизайн Seminar Пушин + преподаватель1
Дизайн Seminar Пушин + преподаватель1
Дизайн Seminar Пушин + преподаватель2
Дизайн Seminar Пушин + преподаватель2
Дизайн Seminar Пушин + преподаватель3
Дизайн Seminar Пушин + преподаватель3
ИнЯз Lecture Англичане
КБ ComputerLab Котов
КБ ComputerLab Котов
Матанализ Lecture Алексеева Ульяна
Матанализ Lecture Алексеева Ульяна
Матанализ Lecture Симонов Иван
Матанализ Lecture Симонов Иван
Матанализ Seminar Алексеева Ульяна
Матанализ Seminar Алексеева Ульяна
Матанализ Seminar Бондарь
Матанализ Seminar Бондарь
Матанализ Seminar Волков
Матанализ Seminar Волков
Матанализ Seminar Липин Антон
Матанализ Seminar Липин Антон
Матанализ Seminar Никифорова
Матанализ Seminar Никифорова
Матанализ Seminar Першаков
Матанализ Seminar Першаков
Матанализ Seminar Симонов Иван
Матанализ Seminar Симонов Иван
Матанализ Seminar Филатова
Матанализ Seminar Филатова
Матстат Lecture Хлопин
Матстат Seminar Стихин
Матстат Seminar Стихин
Машинное обучение Lecture Стихин
Машинное обучение Lecture Стихин
ОП ComputerLab Ермаков
ОП ComputerLab Кулешова Ирина
ОП ComputerLab Маценко
ОП ComputerLab Петряшов
ОП ComputerLab Политов
ОП ComputerLab Романюк
ОП ComputerLab Федоров
ОП ComputerLab Хапов
ОП Seminar Ермаков
ОП Seminar Кулешова Ирина
ОП Seminar Маценко
ОП Seminar Петряшов
ОП Seminar Политов
ОП Seminar Романюк
ОП Seminar Федоров
ОП Seminar Хапов
Питон ComputerLab Анкудинов Александр
Питон ComputerLab Берсенев
Питон ComputerLab Зверев Владимир
Питон ComputerLab Мясников Даниил
Публичные выступления Lecture Свалова Дарья
Создание продукта Lecture Алексей Кулаков
Физкультура Lecture Физруки1
Физкультура Lecture Физруки1
Физкультура Lecture Физруки2
Физкультура Lecture Физруки2
ФП ComputerLab Акентьев Евгений
ФП ComputerLab Войцишевский Антон
ФП ComputerLab Кирпичников Алексей
ФП ComputerLab Неволин Роман Андреевич
Not placed:

